lib/package-managers/npm.js

```diff
@@ -6,25 +6,8 @@ const Promise = require('bluebird');
 const versionUtil  = require('../version-util.js');
 const spawn        = require('spawn-please');
 const packageJson  = require('package-json');
-const ProxyAgent   = require('https-proxy-agent');
-const npmConf      = require('npm-conf')();
-const proxyUrl     = require('get-proxy')();
-
-let agent;
-if (proxyUrl) {
-    const proxy = new URL(proxyUrl);
-    agent = new ProxyAgent({
-        host: proxy.hostname,
-        port: proxy.port,
-        secureProxy: proxy.protocol === 'https:',
-        keepAlive: true,
-        maxSockets: npmConf.get('maxsockets') || 50, // https://docs.npmjs.com/misc/config#maxsockets
-        ca: npmConf.get('ca'),
-        key: npmConf.get('key'),
-        cert: npmConf.get('cert'),
-        rejectUnauthorized: npmConf.get('strict-ssl')
-    });
-}
+
+const {proxyAgent, noProxyCheck} = require('./npm-proxy');
```

Comment:
Exporting two tightly-coupled variables seems like could be improved by a single export. If they are always used in tandem, the API should reflect this. I would suggest exporting a single `proxyAgent` function that returns the `agent` or `null` is NO_PROXY and optionally takes a `scope` as an argument.  `packageName.includes('/') && packageName.split('/')[0] ` would be in `npm.js`. What do you think?

Reply:
Makes sense. I've addressed the proposed changes (which also make the tests easier to understand).